spyder/plugins/ipythonconsole/plugin.py

```diff
@@ -624,7 +624,8 @@ def set_working_directory(self, dirname):
         """Set current working directory.
         In the workingdirectory and explorer plugins.
         """
-        self.sig_current_directory_changed.emit(dirname)
+        if not self.get_current_client().external_kernel:
+            self.sig_current_directory_changed.emit(dirname)

```

Comment:
I think the right fix is the following:

```suggestion
        if osp.isdir(dirname):
            self.sig_current_directory_changed.emit(dirname)
```

That's because an external kernel can be started in the same machine as Spyder. In that case, it won't have problems setting the cwd using this signal.

Reply:
Fixed.